cogent/manual/cogent-manual.tex

```diff
@@ -1230,12 +1237,11 @@ \subsubsection{Variant Patterns}
 A variant pattern conforms to every variant type which has atleast a variant with the \textit{DataConstructor} as its tag. Although a variant 
 pattern matches all values of the type having only that variant, this is not true for all other conforming types. For those types the pattern 
 only matches the subset of value sequences which have been constructed with the \textit{DataConstructor} as its discriminating tag. Therefore variant 
-patterns are always refutable.  As usual, when matched with a value, the match must be exhaustive, specifying
-a pattern for every variant. 
-
-
-When a variant pattern is (successfully) matched with a value, the subpatterns are matched with the remaining values after removing the discriminating tag.
+patterns are always refutable, even if all existing conforming types have only one variant.
+As usual, when matched with a value, the match must be exhaustive, specifying a pattern for every variant.
```

Comment:
In general I find this `Variant Pattern` subsection confusing.

1. "Variant" is ambiguous. It can mean one alternative in a variant type, or a variant type itself. I tend to use variant in the latter sense. I usually describe it as an alternative.
2. "Value" is ambiguous, It sometimes means the payload (i.e. the parameters a data constructor takes), and it sometimes means a value as used in language semantics.
3. "Variant pattern" is not very intuitive and doesn't make much sense to me. (Of course it's related to my understanding of "variant" as in 1.) The pattern is formed of a data constructor and payload.

Reply:
Ok, you are right, too many variants of "variant" 8^).
I hope that it is clearer now.